source/common/http/conn_manager_impl.cc

```diff
@@ -1247,23 +1246,17 @@ void ConnectionManagerImpl::startDrainSequence() {
   drain_timer_->enableTimer(config_.drainTimeout());
 }
 
-bool ConnectionManagerImpl::ActiveStream::snapScopedRouteConfig() {
+void ConnectionManagerImpl::ActiveStream::snapScopedRouteConfig() {
   ASSERT(request_headers_ != nullptr,
          "Try to snap scoped route config when there is no request headers.");
 
-  snapped_route_config_ = snapped_scoped_routes_config_->getRouteConfig(*request_headers_);
   // NOTE: if a RDS subscription hasn't got a RouteConfiguration back, a Router::NullConfigImpl is
   // returned, in that case we let it pass.
+  snapped_route_config_ = snapped_scoped_routes_config_->getRouteConfig(*request_headers_);
   if (snapped_route_config_ == nullptr) {
     ENVOY_STREAM_LOG(trace, "can't find SRDS scope.", *this);
-    // Stop decoding now.
-    maybeEndDecode(true);
-    sendLocalReply(Grpc::Common::hasGrpcContentType(*request_headers_), Http::Code::NotFound,
-                   "route scope not found", nullptr, is_head_request_, absl::nullopt,
-                   StreamInfo::ResponseCodeDetails::get().RouteConfigurationNotFound);
-    return false;
+    snapped_route_config_ = std::make_shared<Router::NullConfigImpl>();
```

Comment:
It would be ideal to include the "route scope not found" body in the response from the router.

Reply:
leave a TODO, I am afraid there isn't a trivial way to accomplish that, we either need to keep a sate in streamInfo and make router filter read from it, Or we need something bigger that I have no clue yet.

Also note I have another TODO (issue #8076), which will add some stats for RDS/SRDS, which should help us to understand/monitor this kinda of no-route error. Though I understand it will not help to tell the difference from the client side.